src/context/ServiceInfoContext.tsx

```diff
@@ -200,6 +208,60 @@ const ServiceInfoProvider = ({ children }: PropsWithChildren<{}>) => {
     [currentWallet, clearCurrentWallet],
   )
 
+  // Fetch rescan progress when rescanning is true
+  useEffect(() => {
+    if (!serviceInfo?.rescanning || !currentWallet) return
+
+    const abortCtrl = new AbortController()
+    // Flag to track if the API is supported
+    let isRescanProgressApiSupported = true
+
+    const fetchRescanProgress = async (): Promise<void> => {
+      // Skip if API is not supported (returned 404 previously)
+      if (!isRescanProgressApiSupported) return
+
+      try {
+        const res = await Api.getRescanInfo({
+          signal: abortCtrl.signal,
+          token: currentWallet.token,
+          walletFileName: currentWallet.walletFileName,
+        })
+
+        if (res.ok) {
+          const data = await res.json()
+          if (!abortCtrl.signal.aborted && data.progress !== undefined) {
+            dispatchServiceInfo({
+              rescanProgress: data.progress,
+            })
+          }
+        } else if (res.status === 404) {
```

Comment:
actually got an "405 Method Not Allowed" on older version. Should we just skip the status check and do it for every `!ok`?
```suggestion
        } else {
```

False positives (e.g. temporary failures) would simply lead to the progress not being displayed, which is okay.